ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/validator/preexecution/PreExecutingValidatingCommitterSpec.scala

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.daml.ledger.validator.preexecution
+
+import java.time.Instant
+
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.DamlStateKey
+import com.daml.ledger.participant.state.kvutils.Raw
+import com.daml.ledger.participant.state.kvutils.`export`.{
+  LedgerDataExporter,
+  SubmissionAggregator,
+  SubmissionInfo,
+}
+import com.daml.ledger.participant.state.v1.SubmissionResult
+import com.daml.ledger.validator.TestHelper.{FakeStateAccess, aParticipantId}
+import com.daml.ledger.validator.reading.StateReader
+import com.daml.ledger.validator.{LedgerStateOperations, LedgerStateWriteOperations}
+import com.daml.lf.data.Ref.ParticipantId
+import com.daml.logging.LoggingContext
+import org.mockito.{ArgumentMatchersSugar, MockitoSugar}
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AsyncWordSpec
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class PreExecutingValidatingCommitterSpec
+    extends AsyncWordSpec
+    with Matchers
+    with MockitoSugar
+    with ArgumentMatchersSugar {
+
+  import PreExecutingValidatingCommitterSpec._
+  import PreExecutionTestHelper._
+
+  "commit" should {
+
+    "write the transactions when no post execution conflicts are detected" in {
+      val fixture = new ValidatingCommitterFixture
+      fixture.validatorReturnsSuccess
+      when(
+        fixture.conflictDetector.detectConflicts(
+          any[PreExecutionOutput[TestReadSet, TestWriteSet]],
+          any[StateReader[DamlStateKey, TestValue]],
+        )(any[ExecutionContext])
+      )
+        .thenReturn(Future.unit)
+      val submissionAggregator = mock[SubmissionAggregator]
+      when(fixture.ledgerDataExporter.addSubmission(any[SubmissionInfo]))
+        .thenReturn(submissionAggregator)
+      when(
+        fixture.postExecutionWriter.write(any[TestWriteSet], any[LedgerStateWriteOperations[Long]])(
+          any[ExecutionContext]
+        )
+      )
+        .thenReturn(Future.successful(SubmissionResult.Acknowledged))
+      fixture.committer
+        .commit(
+          aParticipantId,
+          "corid",
+          Raw.Envelope.empty,
+          Instant.now(),
+          new FakeStateAccess(mock[LedgerStateOperations[Unit]]),
+        )
+        .map(result => {
+          verify(fixture.postExecutionWriter)
+            .write(any[TestWriteSet], any[LedgerStateWriteOperations[Long]])(any[ExecutionContext])
+          verify(submissionAggregator).finish()
+          result shouldBe SubmissionResult.Acknowledged
+        })
```

Comment:
Tiny style recommendation.

```suggestion
        .map { result =>
          verify(fixture.postExecutionWriter)
            .write(any[TestWriteSet], any[LedgerStateWriteOperations[Long]])(any[ExecutionContext])
          verify(submissionAggregator).finish()
          result shouldBe SubmissionResult.Acknowledged
        }
```